src/ci/scripts/setup-environment.sh

```diff
@@ -8,6 +8,13 @@ IFS=$'\n\t'
 
 source "$(cd "$(dirname "$0")" && pwd)/../shared.sh"
 
+# Since matrix variables are readonly in Azure Pipelines, we take
+# INITIAL_RUST_CONFIGURE_ARGS and establish RUST_CONFIGURE_ARGS
+# which downstream steps can alter
+if [[ -v INITIAL_RUST_CONFIGURE_ARGS ]]; then
+    ciCommandSetEnv RUST_CONFIGURE_ARGS "${INITIAL_RUST_CONFIGURE_ARGS}"
+fi
```

Comment:
```suggestion
# Since matrix variables are readonly in Azure Pipelines, we take
# INITIAL_RUST_CONFIGURE_ARGS and establish RUST_CONFIGURE_ARGS
# which downstream steps can alter
# MacOS ships with Bash 3.16, which means we cannot use
# if [[ -v FOO ]], as that was introduced in Bash 4.2
if [[ -z "${INITIAL_RUST_CONFIGURE_ARGS+x}" ]]; then
    INITIAL_RUST_CONFIG=""
    echo "No initial Rust Configure Args set"
else
    INITIAL_RUST_CONFIG="${INITIAL_RUST_CONFIGURE_ARGS}"
    ciCommandSetEnv RUST_CONFIGURE_ARGS "${INITIAL_RUST_CONFIG}"
fi
```

Reply:
Confirmed through testing that this block works with Linux, Mac, and Windows builds!

Reply:
Shamelessly stealing now!